.travis.yml

```diff
@@ -16,15 +24,22 @@ before_install:
   - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16 -extension RANDR"
 
 before_script:
-  - npm install -g jpm mozilla-download
+  - pwd
+  - ls node_modules || true
   - npm install
+  - ls node_modules || true
+  - ls node_modules/.bin || true
+  - ls ~/node_modules || true
+  - npm config ls -l
+  - find / -name node_modules
+  - export JPM_FIREFOX_BINARY=`which firefox`
+  - firefox -v
+  - echo $JPM_FIREFOX_BINARY
   - make all
   - cd ..
-  - mozilla-download --branch mozilla-central --product firefox firefox
-  - export JPM_FIREFOX_BINARY=$TRAVIS_BUILD_DIR/../firefox/firefox
   - cd $TRAVIS_BUILD_DIR/
 
 script:
-  - pwd && ls && cd $TRAVIS_BUILD_DIR/build/addon && jpm test -v
+  - pwd && ls && cd $TRAVIS_BUILD_DIR/build/addon && $TRAVIS_BUILD_DIR/node_modules/.bin/jpm test -v
```

Comment:
I'd probably start by testing a "script" in package.json that says:

```js
"jpm:test": "jpm test -v"
```

Then in the .travis.yml file, see if you can change this line to:
```sh
pwd && ls && cd $TRAVIS_BUILD_DIR/build/addon && npm run jpm:test
```